File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		
		BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
		
		String[] str = in.readLine().split(" ");
		int m = Integer.parseInt(str[0]);
		int n = Integer.parseInt(str[1]);
		
		String[][] chess = new String[m][n];
		
		for (int i=0; i<m; i++) {
			str = in.readLine().split("");
			chess[i] = str;
		}
		in.close();
		
		String[] bw = {"B","W"};
		
		int min = 64;
		int count; 
		
		for (int i = 0; i<m-7; i++) {
			for (int j = 0; j<n-7; j++) {
				count = 0;
				for (int k = i; k<i+8; k++) {
					for (int l = j; l<j+8; l++) {
						//System.out.println(bw[(k+l)%2]);
						if (chess[k][l].equals(bw[(k+l)%2]) == false) {
							count++;
						}
					}
				}
				if (count < min) min = count;
				
				count = 0;
				for (int k = i; k<i+8; k++) {
					for (int l = j; l<j+8; l++) {
						//System.out.println(bw[(k+l+1)%2]);
						if (chess[k][l].equals(bw[(k+l+1)%2]) == false) {
							count++;
						}
					}
				}
				if (count < min) min = count;
			}
		}
		System.out.println(min);
	}
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
import sys

n, m = map(int, sys.stdin.readline().split())

board = []
for i in range(n):
    board.append(list(sys.stdin.readline().rstrip()))

min_count = sys.maxsize
bw_seq = ["B", "W"]

for i in range(n-7):
    for j in range(m-7):

        #bw 순서 검사
        count = 0
        for x in range(i, i+8):
            for y in range(j, j+8):
                if board[x][y] != bw_seq[(x+y)%2]:
                    count += 1
        min_count = min(min_count, count)

        #wb 순서 검사
        count = 0
        for x in range(i, i + 8):
            for y in range(j, j + 8):
                if board[x][y] != bw_seq[(x + y +1) % 2]:
                    count += 1
        min_count = min(min_count, count)
print(min_count)